### PR TITLE
fix(clerk-js): Set card error on prepare code verification

### PR DESCRIPTION
**Pull request by desiprisg**
## Description

Screenshot 2024-04-24 at 16 41 33

## Checklist

- [ ] `npm test` runs as expected.
- [ ] `npm run build` runs as expected.
- [ ] (If applicable) [JSDoc comments](https://jsdoc.app/about-getting-started.html) have been added or updated for any package exports
- [ ] (If applicable) [Documentation](https://github.com/clerk/clerk-docs) has been updated

## Type of change

- [ ] 🐛 Bug fix
- [ ] 🌟 New feature
- [ ] 🔨 Breaking change
- [ ] 📖 Refactoring / dependency upgrade / documentation
- [ ] other:

### CODE REVIEW

File: packages/clerk-js/src/ui/components/SignUp/SignUpPhoneCodeCard.tsx [hunk 1]
```diff
@@ -1,24 +1,33 @@
...
   const phoneVerificationStatus = signUp.verifications.phoneNumber.status;
   const shouldAvoidPrepare = !signUp.status || phoneVerificationStatus === 'verified';
   const prepare = () => {
     if (shouldAvoidPrepare) {
       return;
     }
-    return signUp.preparePhoneNumberVerification({ strategy: 'phone_code' });
+    return signUp
+      .preparePhoneNumberVerification({ strategy: 'phone_code' })
+      .catch(err => handleError(err, [], card.setError));
```
Comment (anagstef) [inline packages/clerk-js/src/ui/components/SignUp/SignUpPhoneCodeCard.tsx:20]: Same as above!

File: .vscode/settings.json [hunk 2]
```diff
@@ -3,7 +3,7 @@
   "typescript.enablePromptUseWorkspaceTsdk": true,
   "typescript.tsdk": "node_modules/typescript/lib",
   "editor.codeActionsOnSave": {
-    "source.fixAll.eslint": true,
-    "source.organizeImports": true
+    "source.organizeImports": "explicit",
+    "source.fixAll.eslint": "explicit",
```
Comment (anagstef) [inline .vscode/settings.json:7]: Is this intentional?

File: packages/clerk-js/src/ui/components/SignUp/SignUpEmailCodeCard.tsx [hunk 3]
```diff
@@ -13,12 +16,19 @@ export const SignUpEmailCodeCard = () => {
     if (shouldAvoidPrepare) {
       return;
     }
-    return signUp.prepareEmailAddressVerification({ strategy: 'email_code' });
+    return signUp
+      .prepareEmailAddressVerification({ strategy: 'email_code' })
+      .catch(err => handleError(err, [], card.setError));
```
[the remark this answers is not in the document]
Comment (desiprisg) [reply packages/clerk-js/src/ui/components/SignUp/SignUpEmailCodeCard.tsx:21]: TS complains. Not sure about how `useFetch` is supposed to work. Maybe @panteliselef can help as this was a duplication before this PR as well.

File: .vscode/settings.json (hunk 2 above)
Comment (desiprisg) [reply .vscode/settings.json:7]: yes

**Review by dimkl — APPROVED** [review]

**Review by nikosdouvlis — APPROVED** [review]

**Pull request merged**